src/adapter/types.ts

```diff
@@ -127,6 +127,8 @@ export type CustomInputValidator<T extends EndpointGenerics> = (
   adapterSettings: T['Settings'],
 ) => AdapterError | undefined
 
+export type CustomOutputValidator = (output: Readonly<AdapterResponse>) => AdapterError | undefined
```

Comment:
I think we could make this use the response type from the Transport, which should avoid needing to do the `as LwbaResponseDataFields['Data']` when it is defined

```suggestion
export type CustomOutputValidator<T extends ResponseGenerics> = (output: T) => AdapterError | undefined
```